zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+the network upgrade, so that miners and mining pools may gradually migrate from their
+existing transparent addresses to Sapling addresses. This also simplifies the consensus
+rules, because the Founders' Reward targets transparent addresses, and thus it remains
+necessary for the time being to support them. A future ZIP could require all coinbase to
+be shielded immediately.
+
+Enforcing coinbase maturity at the consensus level for Sapling outputs would incur
+significant complexity in the consensus rules, because it would require special-casing
+coinbase note commitments in the Sapling commitment tree. The standard recommendation when
+spending a note is to select an anchor 10 blocks back from the current chain tip, which
+acts as a de-facto 10-block maturity on all notes, coinbase included. This might be
+proposed as a consensus rule in future.
+
+There is another reason for shielded coinbase maturity being unnecessary: when a rollback
+occurs that would cause a shielded coinbase output to disappear, it will also invalidate
+every shielded transaction that uses an anchor descending from the tree that the shielded
+coinbase output had been appended to. That is, all economic activity would be rolled back
+in addition to the shielded coinbase output disappearing, so there is no reason to make
+shielded coinbase a special case when the same behaviour occurs in regular shielded notes
+already. In the transparent coinbase case, only direct child transactions of the
+transparent coinbase would become invalid, and thus it would be possible to end up in a
+situation where a logical child transaction (for example, a mining pool paying out miners)
+persists in the block chain after its logical parent (the mining pool receiving a block)
+disappears.
+
+Requiring that note commitments are valid when recovering using a fixed outgoing viewing
+key implies that target addresses and values for all Sapling outputs within the coinbase
+are revealed. This would be necessary to correctly enforce shielded Founders' Reward or
+funding stream outputs, and it is simpler to enforce this on all outputs. Additionally,
+this maintains the ability for network observers to track miners and mining pools.
+Meanwhile, the miners and mining pools could put useful or identifying text in the memo
+fields of the outputs, instead of storing it ad-hoc elsewhere in the coinbase transaction.
+
+
+Security and Privacy Considerations
+===================================
+
+Sapling outputs in coinbase transactions are by design publicly viewable, in contrast to
+Sapling outputs in normal transactions. This does not introduce any privacy regressions,
+because coinbase output values and recipient addresses have always been public
+information.
+
+Revealing the coinbase output notes does not enable anyone else to detect when the note is
+spent, which removes the need for a separate shielding step like is enforced for
+transparent coinbase outputs.
```

Comment:
UX: It might be hard to communicate the subtle risk that this is revealing their address. Knowing the miner's address lets you send dust attacks to them and probably some post-quantum attacks on their privacy, so it might matter to some miners.

Reply:
Added a note about this to the "Security and Privacy Considerations" section.